### PR TITLE
Review chapter: "Limited Types"

### PR DESCRIPTION
**Pull request by gusthoff**
Various improvements and corrections to the chapter.

### CODE REVIEW

File: content/courses/advanced-ada/parts/resource_management/limited_types.rst [hunk 1]
```diff
@@ -537,6 +537,28 @@ actually provide assignment. But the partial view can restrict what is actually
...
+
+    private
+
+       type Rec is tagged record
+          I : Integer;
+       end record;
+
+    end Simple_Recs;
+
+Here, compilation fails because the :ada:`Ref` is nonlimited in its full view.
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/parts/resource_management/limited_types.rst:558]: Good catch, thanks! (Yes, the word "type" was missing there.)

File: content/courses/advanced-ada/parts/resource_management/limited_types.rst [hunk 2]
```diff
@@ -649,6 +671,86 @@ example and try to compile it.)
...
+
+       type Rec is record
+          I : Integer;
+       end record;
+
+    end Simple_Recs;
+
+Now, even though the :ada:`Rec` type was declared as limited private, the full
+view indicates that it's actually a non-limited type, so it isn't explicitly
+limited.
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/parts/resource_management/limited_types.rst:740]: Yes, I added some comments to the previous section that introduces the topic.

File: content/courses/advanced-ada/parts/data_types/types.rst [hunk 3]
```diff
@@ -1617,83 +1617,152 @@ subprograms of the package body that make use of this type. Again, the partial
...
+
+    with Private_Integers; use Private_Integers;
+
+    procedure Show_Private_Integers is
+       A, B : Private_Integer;
+    begin
+       A := A + B;
+    end Show_Private_Integers;
+
+In this example, we use the :ada:`+` operator as a common integer variable.
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/parts/data_types/types.rst]: Yes, this whole sentence sounds really weird.

File: content/courses/advanced-ada/parts/data_types/types.rst [hunk 4]
```diff
@@ -1617,83 +1617,152 @@ subprograms of the package body that make use of this type. Again, the partial
...
+arrays or even discriminants for arrays. For example, the following
+declarations won't work:
+
+.. code:: ada compile_button manual_chop project=Courses.Advanced_Ada.Data_Types.Types.Type_View.Private_Array
+    :class: ada-expect-compile-error
+
+    !private_arrays.ads
+    package Private_Arrays is
+
+       type Private_Unconstrained_Array is private;
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/parts/data_types/types.rst:1717]: These are very good points. I've just integrated this explanation (almost verbatim) to the section.

File: content/courses/advanced-ada/parts/resource_management/limited_types.rst [hunk 5]
```diff
@@ -848,16 +1065,25 @@ procedure to a new procedure in the body of the :ada:`Simple_Recs.Ext` package:
...
+:ada:`Test_Limitedness` procedure, compilation fails there (because
+:ada:`Rec_Derived` is viewed as descending from a limited type).
 
 
 Deriving from tagged limited private types
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The rules for deriving from tagged limited private types are slightly different
-than ones we've seen in the previous section. Let's look at an example:
+than the ones we've seen so far. This is because tagged limited types are
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/parts/resource_management/limited_types.rst]: Thanks a lot for the hint!

**Review by pat-rogers — APPROVED** [review]

**Review by pat-rogers — APPROVED** [review]

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**